Lib/email/_policybase.py

```diff
@@ -14,6 +15,14 @@
     'compat32',
     ]
 
+# equivalent to pattern re.compile("[!-9;-~]+$")
+valid_header_name_re = re.compile("[\041-\071\073-\176]+$")
+
+def validate_header_name(name):
```

Comment:
Actually, I just observed that we also have `email.utils`. I don't know whether this one is better to be placed here or in the utils, next to `_has_surrogates`. @bitdancer any thoughts? Also, we seem to use underscore names for private utils even if the module is already private so maybe `_validate_header_name` can be used as well (this is just a suggestion, but I'd like to hear others' thoughts)

Reply:
Well, the function has no leading underscore because it is a "public" function of the `_policybase` module; that is, it is part of the `_policybase` API and won't change.  But it is not a *public* function because `_policybase` is not public/documented.  That said, it wouldn't be crazy to put it in utils and document it.  But let's not do that absent an actual use case.  `_has_surrorages` is in utils because lots of other modules use it.  `validate_header_name` should, in theory, be used only by modules derived from `_policybase`, since the policy is what determines whether a header name is valid.  I'm inclined to leave it here absent a use case, and since it is private it can be more easily moved later if there  is a need.